Source/Core/Core/IOS/Network/Socket.cpp

```diff
@@ -712,24 +715,81 @@ void WiiSockMan::Update()
       socket_iter = WiiSockets.erase(socket_iter);
     }
   }
-  s32 ret = select(nfds, &read_fds, &write_fds, &except_fds, &t);
+  s32 ret = poll(ufds.data(), static_cast<u32>(ufds.size()), 0);
+  if (ret < 0)
+  {
+    ERROR_LOG(IOS_NET, "SO_POLL poll failed (%d)", GetNetErrorCode(ret, "Update", true));
+    return;
+  }
 
-  if (ret >= 0)
+  int n = 0;
+  for (auto& elem : WiiSockets)
   {
-    for (auto& pair : WiiSockets)
-    {
-      WiiSocket& sock = pair.second;
-      sock.Update(FD_ISSET(sock.fd, &read_fds) != 0, FD_ISSET(sock.fd, &write_fds) != 0,
-                  FD_ISSET(sock.fd, &except_fds) != 0);
-    }
+    elem.second.Update(!!(ufds[n].revents & read_event), !!(ufds[n].revents & write_event),
+                       !!(ufds[n].revents & error_event));
+    n++;
   }
-  else
+
+  UpdatePollCommands();
+}
+
+void WiiSockMan::UpdatePollCommands()
+{
+  constexpr static int error_event = (POLLHUP | POLLERR);
```

Comment:
The static isn't necessary here. Given it's a trivial primitive type, this will be constant folded into expressions anyway (especially given it's constexpr).

Our convention is also to place the lifetime specifiers before mutability ones, just to keep in mind for the future.

Reply:
I know, but it won't [build on Windows](https://dolphin.ci/#/builders/21/builds/948/steps/3/logs/stdio) without it due to a compiler bug.